control-plane/pkg/reconciler/consumer/controller.go

```diff
@@ -90,7 +91,21 @@ func NewController(ctx context.Context, watcher configmap.Watcher) *controller.I
 		impl.GlobalResync(consumerInformer.Informer())
 	}
 
-	cgreconciler.ResyncOnStatefulSetChange(ctx, globalResync)
+	cgreconciler.ResyncOnStatefulSetChange(ctx, impl.FilteredGlobalResync, consumerInformer.Informer(), func(obj interface{}) (*kafkainternals.ConsumerGroup, bool) {
+		c, ok := obj.(*kafkainternals.Consumer)
+		if !ok {
+			return nil, false
+		}
+
+		for _, ref := range c.GetOwnerReferences() {
+			if ref.Kind == "ConsumerGroup" {
```

Comment:
I'd use case insensitive match
```suggestion
			if strings.EqualFold(ref.Kind, "ConsumerGroup") {
```

Reply:
(Make sure the import is there)

Reply:
Actually, there is this function

https://github.com/knative-extensions/eventing-kafka-broker/blob/0c5c2e542d0d9f26bcd034c8f0da8fa38872616f/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1/consumer_types.go#L258-L267